File: js/armstrongNumbersSpec.js
// Can you translate this driver code to unit tests?

var arm = require("./armstrongNumbers");
var shallowEqualArrays = require('shallow-equal/arrays');

function createArrayOfNum(maxValue) {
  return Array.apply(null, {length: maxValue}).map(Number.call, Number)
}

console.log(shallowEqualArrays(arm.findArmstrongNumbers([0]),[0]));
console.log(shallowEqualArrays(arm.findArmstrongNumbers(createArrayOfNum(8)), [0, 1, 2, 3, 4, 5, 6, 7]));
console.log(shallowEqualArrays(arm.findArmstrongNumbers(createArrayOfNum(99)), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]));
console.log(shallowEqualArrays(arm.findArmstrongNumbers(createArrayOfNum(999)), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 153, 370, 371, 407]));

File: python/armstrong_numbers_spec.py
# The code is written as driver code. Can you convert it to Unit Test?
from armstrong_numbers import find_armstrong_numbers
import unittest

class TestArmstrong(unittest.TestCase):

    def test_armstrong(self):
        self.assertEqual(find_armstrong_numbers([0]), [0])
        self.assertEqual(find_armstrong_numbers(list(range(0, 8))), [0, 1, 2, 3, 4, 5, 6, 7])
        self.assertEqual(find_armstrong_numbers(list(range(0, 100))), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
        self.assertEqual(find_armstrong_numbers(list(range(0, 999))), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 153, 370, 371, 407])


if __name__ == '__main__':
    unittest.main()


# print(find_armstrong_numbers([0]) == [0]) 
# print(find_armstrong_numbers(list(range(0, 8))) == [0, 1, 2, 3, 4, 5, 6, 7])
# print(find_armstrong_numbers(list(range(0,100))) == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
# print(find_armstrong_numbers(list(range(0,999))) == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 153, 370, 371, 407])
